### ISSUE TITLE
Needs more documentation regarding Modals

### ISSUE THREAD

**Issue by echosalik** [krescruz/angular-materialize#185]
There is little or no detail on how modal works. Please give more details regarding modals like a code sample etc. Not just HTML markup.

**Comment by webbiesdk** [comment]
2 things: 
- index.html works as the code sample. It showcases almost every single feature. Not just with the code-samples shown, but also the source-code of index.html itself. (And in the modal example, that includes how the callbacks work). 
- The best way to get more details on how the modals work, is to write it yourself. Anyone can send in a pull-request, and a lot of people have. Just the modal directive have 5 different authors according to git blame. 

Remember, no-one is getting paid to do this.

**Comment by echosalik** [comment]
Dude, if i was that good with angular I would have submitted a pull request. :P Plus thanks for the direction. I was using developer tools to view the HTML and wasn't able to find it.

**Comment by echosalik** [comment]
@webbiesdk would you be abel to provide a modal example with controller? All i see is this line in ModalController `scope.openModal = false;` but I dont get it how to use it. Like where is openModal applied etc.

**Comment by echosalik** [comment]
Ok, I got it to work. Which is kinda a long route to be honest. You need an extra element for this to work. As you have already labeled it as an enhancement, my suggestion for the enhancement is: **Allow modals to be set without this extra element (trigger element)**.

**Comment by mushtaqak** [comment]
I agree with @webbiesdk that if we want more documentation then we should need to add some documentation ourselves, this way future developers won't have to waste a lot of times. 👍

**Comment by vikasraok** [comment]
@DragoTechInnovations put up your solution here maybe? I just went with the ui.bootstrap modal instead.
